axelrod/fingerprint.py

```diff
@@ -1,60 +1,58 @@
+"""
+changelog:
+
+added/changed docstring
+
+- AshlockFingerprint - added summary
+- fingerprint - step added one line
+- fingerprint - added missing params
+- construct_tournament_elements - step added one sentence
+- create_jossann - probe param now class or instance not class
+
+code:
+
+create_points - `num = int((1 / step) // 1) + 1` to `num = int(1 / step) + 1`
+plot - `size = int((1 / self.step)) // 1 + 1` to `size = int(1 / self.step) + 1`
+__init__ - explicitly created all the class values and assigned to None.
+
+"""
+
+
 from collections import namedtuple
 from tempfile import NamedTemporaryFile
+from typing import Tuple, List, Dict, Union, Any
 import matplotlib.pyplot as plt
 import numpy as np
 import tqdm
 import axelrod as axl
 
-from axelrod import on_windows
+from axelrod import Player, SpatialTournament
+from axelrod.actions import Action
 from axelrod.strategy_transformers import JossAnnTransformer, DualTransformer
 from axelrod.interaction_utils import (
     compute_final_score_per_turn, read_interactions_from_file)
 
 
 Point = namedtuple('Point', 'x y')
 
+PlayerOrStrategy = Union[Player, type]
```

Comment:
I personally do not like type aliases for type hinting. They aren't helpful when accessing the docstring/syntax of a function from elsewhere. I suggest these just get written verbosely throughout.